### ISSUE TITLE
Provide a health check that would fail if node certificate expires in a certain amount of time

### ISSUE THREAD

**Issue by michaelklishin** [rabbitmq/rabbitmq-cli#305]
See #292 and #304 for background and [Pivotal CF Certificate Rotation](https://docs.pivotal.io/pivotalcf/2-3/security/pcf-infrastructure/api-cert-rotation.html#checks) docs for inspiration.

```
rabbitmq-diagnostics check_certificate_expiration --within 1 --unit month

rabbitmq-diagnostics check_certificate_expiration --within 3 --unit weeks
```

Suggested by @lynhines.

[168224642]

**Comment by codeadict** [comment]
So, this will return non-zero if any listener has a certificate that expires within the given period, probably returning the `listener` and `expires_on`? In case the no certificate expires in that period it returns 0 with a message indicating that there is not any certificate expiring. Should the unit account for days and years as well? If my assumptions are correct I could contribute this command. To my reasoning #304 should be done before this one.

**Comment by michaelklishin** [comment]
@codeadict correct. We can use that format (`30d`, `2m`, `1y` and so on), although anything else that's reasonable also would work.

**Comment by michaelklishin** [comment]
Implemented in #381 by @dcorbacho and @acogoluegnes.

**Comment by michaelklishin** [comment]
HTTP API versions of this check has shipped in https://github.com/rabbitmq/rabbitmq-management/issues/844.
